app/routes/events/view/index.js

```diff
@@ -1,24 +1,33 @@
 import Route from '@ember/routing/route';
+import EmberTableRouteMixin from 'open-event-frontend/mixins/ember-table-route';
 
-export default Route.extend({
-  async model() {
+export default class extends Route.extend(EmberTableRouteMixin) {
+
+  async model(params) {
     let eventDetails = this.modelFor('events.view');
-    let sponsorQueryObject = {
-      'page[size]': 10
+    const searchField = 'name';
+    let filterOptions = [];
+    filterOptions = this.applySearchFilters(filterOptions, params, searchField);
+    let queryString = {
+      filter         : filterOptions,
+      'page[size]'   : params.per_page || 10,
+      'page[number]' : params.per_page || 1
     };
+    queryString = this.applySortFilters(queryString, params);
+
 
     return {
       event             : await eventDetails,
-      sponsors          : await eventDetails.query('sponsors', sponsorQueryObject),
-      query             : sponsorQueryObject,
+      sponsors          : await eventDetails.query('sponsors', queryString),
+      query             : queryString,
       sponsorObjectType : 'sponsors',
       roleInvites       : await eventDetails.query('roleInvites', {}),
       sessionTypes      : await eventDetails.query('sessionTypes', {}),
       socialLinks       : await eventDetails.query('socialLinks', {}),
       statistics        : await eventDetails.query('eventStatisticsGeneral', {}),
       orderStat         : await eventDetails.query('orderStatistics', {}),
       tickets           : await eventDetails.query('tickets', {}),
-      roles             : await this.store.findAll('role')
+      roles             : await this.asArray(this.store.findAll('role'))
```

Comment:
what has the sponsors table got to do with roles?

Reply:
Nothing. So I need to use asArray proxy for sponsors?

Reply:
yes

Reply:
@CosmicCoder96 Please check

Reply:
@CosmicCoder96 Please check, Still, the issue is not solved,